editor/src/components/images.ts

```diff
@@ -54,14 +48,88 @@ export function getImageSrc(
   }
 }
 
-export function parseImageMultiplier(imagePath: string): number {
-  const imageMultiplierRegex = /.*@(\d*)x\..*/
-  const imageMultiplierResult = imageMultiplierRegex.exec(imagePath)
-  let multiplier: number = 1
-  if (imageMultiplierResult != null && imageMultiplierResult.length === 2) {
-    multiplier = Number.parseInt(imageMultiplierResult[1])
+export interface FilenameParts {
+  filename: string
+  extension: string
+  multiplier?: number
+  deduplicationSeqNumber?: number
+}
+
+export function filenameFromParts(parts: FilenameParts): string {
+  const { filename, multiplier, extension, deduplicationSeqNumber } = parts
+  const multiplierString = optionalMap((m) => `@${m}x`, multiplier) ?? ''
+  const dedupeString = optionalMap((n) => `_${n}`, deduplicationSeqNumber) ?? ''
+  return `${filename}${dedupeString}${multiplierString}.${extension}`
+}
+
+export interface LastPartSeparatedByResult<T> {
+  part: T
+  rest: string
+}
+
+function lastPartSeparatedBy<Separator extends string, T>(
+  sepBy: Separator,
```

Comment:
Is there any benefit here to creating a specific type for this?

Reply:
well this can go either way, I created the type so it's easier to see what the function separates along, and so that the first parameter is not a simple string

I'm ok with converting the function into one that takes an object instead of two strings, if that would be an improvement

Reply:
yeah I don't think it adds valuable information to make the Separator a generic type like this

Reply:
thanks for the feedback, I removed the unnecessary generic param